From 13cdca454e25ca5759c277e98cfe7369a07cdbc2 Mon Sep 17 00:00:00 2001
From: simeonwarren <simeonwarren-git-pzewq@alwaldend.com>
Date: Sun, 7 Dec 2025 13:22:03 +0300
Subject: [PATCH 2/5] Modify 30m chimes interval

---
 src/components/datetime/DateTimeController.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/components/datetime/DateTimeController.cpp b/src/components/datetime/DateTimeController.cpp
index d439821b..19109533 100644
--- a/src/components/datetime/DateTimeController.cpp
+++ b/src/components/datetime/DateTimeController.cpp
@@ -117,12 +117,12 @@ void DateTime::UpdateTime(uint32_t systickCounter, bool forceUpdate) {
     isHourAlreadyNotified = false;
   }
 
-  if ((minute == 0 || minute == 30) && !isHalfHourAlreadyNotified) {
+  if ((minute == 0 || minute == 20 || minute == 40) && !isHalfHourAlreadyNotified) {
     isHalfHourAlreadyNotified = true;
     if (systemTask != nullptr) {
       systemTask->PushMessage(System::Messages::OnNewHalfHour);
     }
-  } else if (minute != 0 && minute != 30) {
+  } else if (minute != 0 && minute != 20 && minute != 40) {
     isHalfHourAlreadyNotified = false;
   }
 
-- 
2.51.1

