diff --git a/MODULE.bazel b/MODULE.bazel
index 0fcd082..798bd62 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -16,6 +16,8 @@ bazel_dep(name = "bazel_skylib", version = "1.7.1", dev_dependency = True)
 bazel_dep(name = "rules_cc", version = "0.2.2", dev_dependency = True)
 bazel_dep(name = "rules_java", version = "8.16.1", dev_dependency = True)
 bazel_dep(name = "rules_pkg", version = "1.0.1", dev_dependency = True)
-bazel_dep(name = "rules_python", version = "0.35.0", dev_dependency = True)
+
+bazel_dep(name = "rules_python", version = "0.35.0")
+
 bazel_dep(name = "rules_testing", version = "0.6.0", dev_dependency = True)
 bazel_dep(name = "stardoc", version = "0.6.2", dev_dependency = True)
